packages/hebao_v1/contracts/modules/security/GuardianModule.sol

```diff
@@ -73,45 +74,47 @@ contract GuardianModule is SecurityModule
         require(count < MAX_GUARDIANS, "TOO_MANY_GUARDIANS");
 
         if (controller.securityStore().numGuardians(wallet) == 0) {
-            controller.securityStore().addGuardian(wallet, guardian);
-            emit GuardianAdded(wallet, guardian);
+            controller.securityStore().addGuardian(wallet, guardian, info);
+            emit GuardianAdded(wallet, guardian, info);
         } else {
-            uint confirmStart = pendingAdditions[wallet][guardian];
+            uint confirmStart = pendingAdditions[wallet][guardian][info];
             require(confirmStart == 0 || now > confirmStart + confirmPeriod, "ALREADY_PENDING");
-            pendingAdditions[wallet][guardian] = now + pendingPeriod;
-            emit GuardianAdditionPending(wallet, guardian, now + pendingPeriod);
+            pendingAdditions[wallet][guardian][info] = now + pendingPeriod;
+            emit GuardianAdditionPending(wallet, guardian, info, now + pendingPeriod);
         }
     }
 
     function confirmGuardianAddition(
         address wallet,
-        address guardian
+        address guardian,
+        uint    info
         )
         external
         nonReentrant
         onlyFromMetaTxOrWalletOwner(wallet)
         onlyWhenWalletUnlocked(wallet)
     {
-        uint confirmStart = pendingAdditions[wallet][guardian];
+        uint confirmStart = pendingAdditions[wallet][guardian][info];
         require(confirmStart != 0, "NOT_PENDING");
         require(now > confirmStart && now < confirmStart + confirmPeriod, "EXPIRED");
-        controller.securityStore().addGuardian(wallet, guardian);
-        delete pendingAdditions[wallet][guardian];
-        emit GuardianAdded(wallet, guardian);
+        controller.securityStore().addGuardian(wallet, guardian, info);
+        delete pendingAdditions[wallet][guardian][info];
```

Comment:
A guardian can only be added a single time. I have added code to update the guardian. In the edge case a guardian is added with different `types` values it's still the wallet owner that needs to confirm it in a certain order so that seems okay for me.